gnosis/safe-react
# Added spending limit reset function and call

**What**: 

Fixes #1700 #1698 #2262

**Why**: 

The spent amount didn't update when a new allowance was set

**How**:  

Before setting a new allowance we check if there's a spending. If that's the case we reset the spending.
@rmeissner I saw in the Allowance Module Contract that the setAllowance function sets all the Allowance values except for the spent parameter. Is this intended? Because this allows the contract to have invalid Allowances (e.g.: An allowance with 10 ETH for limit and 20 ETH for spent)

**How to test it**:

1 - Allow an owner 1 of a token (ETH for example)
2 - With the owner use the 0.5 ETH
3 - Edit the Spending limit, from 1ETH to 0.25 ETH
4 - You should see the new spending limit with the spent on 0

**Checklist**:

- [ ] TypeScript definitions updated N/A
